### DISCUSSION TITLE
Proposal of exercise GUI

### DISCUSSION THREAD

**Discussion by dmariaa** [JdeRobot/RoboticsAcademy#1783] [Ideas]
Hi all!! As part of the migration to a React based frontend, I am rethinking the exercise GUI with a Material Design perpective in mind... Here come my proposals, just don't hesitate to give your opinion, alternatives, problems you see... whatever is welcome!

**Comment by dmariaa** [comment]
<img width="810" alt="code_material" src="https://user-images.githubusercontent.com/5584976/180616127-dc578be0-a280-4079-9b6b-186cef0f9975.png">
 
<img width="810" alt="output_material" src="https://user-images.githubusercontent.com/5584976/180616577-32d1aab9-b95e-4698-a4c2-9e2ee600bb57.png">

<img width="810" alt="output_gazebo" src="https://user-images.githubusercontent.com/5584976/180619960-89657de7-2247-435a-916d-734a9893451e.png">


 This is the proposal for the "Follow line" exercise, most of the components are common for all exercises though

**Comment by Apoorvgarg-creator** [comment]
What features are there in the drawer “top left” ?

**Reply by dmariaa** [reply]
I am working on it... Account, links to help for the exercise and forum, back to exercise list ... maybe some settings...
¿Any ideas?

**Reply by dmariaa** [reply]
Something like this...

<img width="808" alt="drawer_menu" src="https://user-images.githubusercontent.com/5584976/180641910-30e030db-72b2-4636-8f3f-ff622cd288c4.png">

**Comment by dmariaa** [comment]
Some interactions and messages...

<img width="1006" alt="tools" src="https://user-images.githubusercontent.com/5584976/180634395-81bead23-9466-4beb-aaaf-23373bcac822.png">

**Comment by dmariaa** [comment]
Functionality to record the exercise output window, usefull for showcasing, students work...

<img width="524" alt="screen_recording" src="https://user-images.githubusercontent.com/5584976/180635344-2aa4ca7e-3301-4c6b-bf85-ff7c45fbde87.png">

The record icon turns red and the dot inside blincks to show up recording is active

<img width="510" alt="screen_recording_active" src="https://user-images.githubusercontent.com/5584976/180635343-1a26c322-1a21-421e-b6a5-d95c5b7511c3.png">

**Comment by dmariaa** [comment]
Modal alert and dialogs...

<img width="811" alt="modal_alert" src="https://user-images.githubusercontent.com/5584976/180641596-3240abe3-5c48-4d24-b470-68bcce39fa21.png">

**Reply by jmplaza** [reply]
Wow awesome GUI evolution @dmariaa !!!

Regarding Brain Frequency the user should set (and see) the **desired** Brain Freq and to see the **measured** Brain Freq. That is, two values instead of the only one appearing in your prototype. For instance, 7/10, where 7 is the currently measured one and 10 is the desired one (which should also be changeable from frontend, with the + and - buttons for instance). The current Brain Freq is iteratively being measured all the time.

The same applies for GUI Frequency.

**Reply by dmariaa** [reply]
Ok, I will redesign the components taking that ideas into account...

**Reply by dmariaa** [reply]
This should be a more proper design...

<img width="283" alt="Captura de Pantalla 2022-08-17 a las 8 34 32" src="https://user-images.githubusercontent.com/5584976/185050980-932c191d-fa77-4d9b-9b2b-35f9439f0c50.png">

**Reply by jmplaza** [reply]
Yeah, clear and informative! The blue color for measured frequency is nice

**Reply by dmariaa** [reply]
Now i realize that the buttons are swapped, plus should be right side, minus should be left side
